pyClient/test/test_signing.py

```diff
@@ -0,0 +1,33 @@
+from zeth import signing
+from hashlib import sha256
+from unittest import TestCase
+from os import urandom
+
+
+class TestSigning(TestCase):
+
+    def test_sign_verify(self) -> None:
+        """
+        Test the correct signing-verification flow:
+        verify(vk, sign(sk,m), m) = 1
+        """
+        m = sha256("clearmatics".encode()).digest()
+        keypair = signing.gen_signing_keypair()
+        sigma = signing.sign(keypair.sk, m)
+        self.assertTrue(signing.verify(keypair.vk, m, sigma))
+
+        keypair2 = signing.gen_signing_keypair()
+        self.assertFalse(signing.verify(keypair2.vk, m, sigma))
+
+    def test_sign_verify_random(self) -> None:
+        """
+        Test the correct signing-verification flow with random message:
+        verify(vk, sign(sk,m), m) = 1
+        """
+        m = urandom(32)
```

Comment:
can you 
- change to a length strictly inferior than 32 (30 for instance)
- change `self.assertTrue(signing.verify(keypair.vk, m, sigma))` -> self.assertFalse(signing.verify(keypair.vk, m, sigma))
- add `self.assertTrue(signing.verify(keypair.vk, M, sigma))` with M being m encoded over 32 bytes ?

Reply:
Sorry, can you explain this?  I don't follow.  It sounds like you are asking for another unit test.  Do you want to check that verification fails for a truncated message digest m[0:30]?  Or that signing fails if the message digest is too short?

Reply:
After thinking a bit more we dont need a unit test for that sorry. My thinking:

There is actually no problem per se.

It is just that on the client side we have a signature scheme which sign a message of any length and on the contract we have a verification function which verifies message of 32 bytes only.

An easy error we could have is that we sign a message which is 31 bytes (should not happen in our case because this is the output of sha256). On the contract side as this variable is cast as a bytes32 it is padded so the variable is different and the verification fails (because we hash the variable).
 
I just thought it would be good to have a reminder somewhere. A check on the message size on the signature algo would restrict it so I do not think it is a good idea. We don't need a unit test for that.